Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ReviewRequestController.java

```diff
@@ -21,12 +24,38 @@
            
     @RequestMapping(value = "/auth/request/add", method = RequestMethod.POST )
     public @ResponseBody ReviewRequest addNewReviewRequest( @ModelAttribute(value="reviewRequest") ReviewRequest reviewRequest,Principal principal) {
-  
-        reviewRequest.setStatus(ReviewStatus.OPENED);
         reviewRequest.setRequester(principal.getName());
-        
+        reviewRequest.setCreatedAt(LocalDateTime.now());
         requestsMgr.addReviewRequest(reviewRequest);
         
         return reviewRequest;
     }
+    
+    @RequestMapping(value = "/auth/request/resolve", method = RequestMethod.POST )
+    public @ResponseBody ReviewRequest resolveRequest(@RequestBody ReviewRequest reviewRequest,Principal principal) {
+        reviewRequest.setResolver(principal.getName());
+        reviewRequest.getComments().get(0).setCreatedBy(principal.getName());
+        reviewRequest.getComments().get(0).setCreatedAt(LocalDateTime.now());
```

Comment:
why are you setting these two properties at the first comment? Are you assuming that the first comment is a new one? ReviewRequest is the data send from the front end, right? So you assume that whatever is being send from the front end is exactly what's in the database plus a resolving comment? You cannot assume that. I could manipulate the data being sent (e.g. deleting all comments) and you would just store it in the database without checking. You can't trust what's coming from the frontend. Also, what if someone else resolved a request in the meantime? You'd loose their resolution comment and resolve it again.

Reply:
Julia, In here I'm assuming that I'll be getting only one comment, since it's tightly coupled with Resolve/Reopen Request. And I'm only appending it to the already present comments for this specific reviewRequest.

Reply:
but that will change when we add a commenting feature.

Reply:
So, I should have a separate rest end point for adding comments and fire two calls(One to save comments and one for changing status) while resolving or reopening request for now ?

Reply:
no. You assume that the review request you're getting from the front end is valid the way you receive it and that it doesn't have more than one comment, right? Two issues with that:
1) You are not validating what's being passed from the front end. I could set the status to whatever I like for example and you would simply store that in the database.
2) the same way, I could either add a bunch of comments to the review and you would store them all, or once we have a commenting feature, the passed review might originally have several comments on them and I could delete them all by sending modified data to the database.

Reply:
Sure. I think now I got the point. I'll keep a check in the controller.